common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -222,6 +227,10 @@ protected void serviceInit(Configuration externalConf) throws Exception {
 
     boolean stopOnFailure = _conf.getBoolean(STOP_ON_FAILURE_KEY, DEFAULT_STOP_ON_FAILURE);
 
+    if (_recoveryPath == null && _conf.getBoolean(INTEGRATION_TESTING, false)) {
+      _recoveryPath = new Path(Files.createTempDir().toURI());
```

Comment:
There is a little difference here. 
`com.google.common.io.Files.createTempDir` is used instead due to  SPARK-39102 is not in Spark 3.3, so there is no `createTempDir` in `JavaUtils`